### PR TITLE
Support juju proxy settings

### PR DESCRIPTION
**Pull request by nrobinaubertin**
## Issue
Fixes https://github.com/canonical/loki-k8s-operator/issues/326

## Solution
I declared a ProxyHandler for the urllib request that is made to download the promtail binary.

## Testing Instructions
I used the discourse-k8s charm to test the change.  
1. Add the cilium addon to microk8s
2. Deploy squid on microk8s like proposed [here](https://hub.docker.com/r/ubuntu/squid)
3. Create a model
4. apply a network-policy to that model only allowing local networking (like [this](https://gist.github.com/nrobinaubertin/f583409700bced7edf03827d3fd48134)) 
5. Apply the correct juju-http{s,}-proxy settings to the model
6. relate all the charms to discourse-k8s, it should be operational in the end.

## Release Notes
Add support for JUJU_CHARM_{HTTP,HTTPS}_PROXY variables in the loki_push_api lib.

### CODE REVIEW

**Review by dstathis — APPROVED** [review]

**Comment by lucabello** [comment]
This PR needs to bump the `LIBPATCH` for both libraries :)

**Review by dstathis — DISMISSED** [review]

**Comment by nrobinaubertin** [comment]
I see that the integration tests are failing but I can't see anything related to my change in the logs. Can you help me on that ?

File: lib/charms/loki_k8s/v0/loki_push_api.py [hunk 1]
```diff
@@ -2115,7 +2115,22 @@ def _download_and_push_promtail_to_workload(self, promtail_info: dict) -> None:
                - "zipsha": sha256 sum of zip file of promtail binary
                - "binsha": sha256 sum of unpacked promtail binary
         """
-        with request.urlopen(promtail_info["url"]) as r:
+        # Check for Juju proxy variables and fall back to standard ones if not set
+        proxies: Optional[Dict[str, str]]
+        if os.environ.get("JUJU_CHARM_HTTP_PROXY") and os.environ.get("JUJU_HTTPS_PROXY"):
```
Comment (dstathis) [inline lib/charms/loki_k8s/v0/loki_push_api.py:2120]:
```suggestion
        if os.environ.get("JUJU_CHARM_HTTP_PROXY") and os.environ.get("JUJU_CHARM_HTTPS_PROXY"):
```

Comment (nrobinaubertin) [reply lib/charms/loki_k8s/v0/loki_push_api.py]: Fixed

**Comment by lucabello** [comment]
What's the status on this one @dstathis ?

**Review by dstathis — APPROVED** [review]

**Pull request merged**